### PR TITLE
Refactor scanner to not use a dummy class

### PR DESCRIPTION
**Pull request by arv**
Previously we had a class for the scanner but most of the
implementation used non instance state for performance. Now, we use
functions that operate on the module local variables.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

You asked for it ;-)

File: src/syntax/Scanner.js [hunk 1]
```diff
@@ -206,109 +206,89 @@ let index, input, length, token, lastToken, lookaheadToken, currentCharCode,
...
+ */
+export function getPosition() {
+  return getPosition2(getOffset());
 }
 
 /**
  * @return {SourcePosition}
  */
-function getPosition(offset) {
+function getPosition2(offset) {
```
Comment (johnjbarton) [inline src/syntax/Scanner.js]: Is `2` significant in some way? Maybe `getPositionByOffset()`?

**Comment by johnjbarton** [comment]
LGTM, any impact on performance?

**Comment by arv** [comment]
I expected it to have higher impact. I got 1% improvement which I'm not even sure is significant.

I also created a branch with instance state to make sure that it is still unacceptable and it still is. It was 50% slower.

File: src/syntax/Scanner.js (hunk 1 above)
Comment (arv) [inline src/syntax/Scanner.js]: 2 as in helper. I like your name.

**Pull request merged**